packages/assets/path-support.js

```diff
@@ -10,83 +10,4 @@
 
 'use strict';
 
-import type {PackagerAsset} from './registry.js';
-
-const androidScaleSuffix = {
-  '0.75': 'ldpi',
-  '1': 'mdpi',
-  '1.5': 'hdpi',
-  '2': 'xhdpi',
-  '3': 'xxhdpi',
-  '4': 'xxxhdpi',
-};
-
-const ANDROID_BASE_DENSITY = 160;
-
-/**
- * FIXME: using number to represent discrete scale numbers is fragile in essence because of
- * floating point numbers imprecision.
- */
-function getAndroidAssetSuffix(scale: number): string {
-  if (scale.toString() in androidScaleSuffix) {
-    return androidScaleSuffix[scale.toString()];
-  }
-  // NOTE: Android Gradle Plugin does not fully support the nnndpi format.
-  // See https://issuetracker.google.com/issues/72884435
-  if (Number.isFinite(scale) && scale > 0) {
-    return Math.round(scale * ANDROID_BASE_DENSITY) + 'dpi';
-  }
-  throw new Error('no such scale ' + scale.toString());
-}
-
-// See https://developer.android.com/guide/topics/resources/drawable-resource.html
-const drawableFileTypes = new Set([
-  'gif',
-  'jpeg',
-  'jpg',
-  'ktx',
-  'png',
-  'svg',
-  'webp',
-  'xml',
-]);
-
-function getAndroidResourceFolderName(
-  asset: PackagerAsset,
-  scale: number,
-): string | $TEMPORARY$string<'raw'> {
-  if (!drawableFileTypes.has(asset.type)) {
-    return 'raw';
-  }
-  const suffix = getAndroidAssetSuffix(scale);
-  if (!suffix) {
-    throw new Error(
-      "Don't know which android drawable suffix to use for scale: " +
-        scale +
-        '\nAsset: ' +
-        JSON.stringify(asset, null, '\t') +
-        '\nPossible scales are:' +
-        JSON.stringify(androidScaleSuffix, null, '\t'),
-    );
-  }
-  return 'drawable-' + suffix;
-}
-
-function getAndroidResourceIdentifier(asset: PackagerAsset): string {
-  return (getBasePath(asset) + '/' + asset.name)
-    .toLowerCase()
-    .replace(/\//g, '_') // Encode folder structure in file name
-    .replace(/([^a-z0-9_])/g, '') // Remove illegal chars
-    .replace(/^assets_/, ''); // Remove "assets_" prefix
-}
-
-function getBasePath(asset: PackagerAsset): string {
-  const basePath = asset.httpServerLocation;
-  return basePath.startsWith('/') ? basePath.slice(1) : basePath;
-}
-
-module.exports = {
-  getAndroidResourceFolderName,
-  getAndroidResourceIdentifier,
-  getBasePath,
-};
+module.exports = require('./src/path-support');
```

Comment:
The `"exports"` field should be fully supported in RN's current version of Node (16.x), and in React Native 0.73 we've raised this to 18.x.

Therefore, it should instead be possible to define additional `"exports"` subpaths that represent these files, without actually having them in the package:

```js
  "exports": {
    "./path-support": "./src/path-support.js",
    "./src/path-support": "./src/path-support.js",
    // ...
  },
```

Also, if we're aware of all dependant packages, we should feel free to adopt `"exports"` fully as a breaking change.

Reply:
Yeah, the reason I opted for this is because we (as in Microsoft) will likely need to support older versions still. It would be great if this package can keep backwards compatibility with 0.68.